kernel/src/main/java/org/kframework/kompile/KompileOptions.java

```diff
@@ -241,7 +241,21 @@ public String syntaxModule(FileUtil files) {
 
   @Parameter(
       names = "--enable-kore-antileft",
-      description = "Enable generation of legacy antileft priority predicates ",
+      description = "Enable generation of legacy antileft priority predicates.",
       hidden = true)
   public boolean enableKoreAntileft;
+
+  public enum TypeInferenceMode {
+    Z3,
+    SIMPLESUB,
+    CHECKED,
+  }
+
+  @Parameter(
+      names = "--type-inference-mode",
+      description =
+          "Choose between the Z3-based and SimpleSub-based type inference algorithms, or run both"
+              + " and check that their results are equal. Must be one of [z3|simplesub|checked].",
+      hidden = true)
+  public TypeInferenceMode typeInferenceMode = TypeInferenceMode.SIMPLESUB;
```

Comment:
The default is currently feature-on so that performance runs are accurate, but this should be switched feature-off before merging.

Reply:
I'm not sure I understand the rationale behind leaving it off by default. Surely we want it on so that it gets tested by our integration testing moving forward, right?

Reply:
@dwightguth CI passes with `CHECKED` mode, so we do infer all the same sorts (including on the performance tests which should cover EVM, Elrond, etc.).

However, given the conceptual issues I mentioned in another comment, I would like a bit more time to either
- address those issues in a follow-up PR, or
- prove to myself that they aren't real issues 

before turning the feature on.

Reply:
After discussing with @dwightguth, we decided to leave it off by default, but ensure it still gets tested to an extent on CI by adding `--type-inference-mode simplesub` in `ktest.mak` for the regression tests.